README.md

```diff
@@ -118,9 +118,30 @@ The script waits up to 120 secs (4 x 30 seconds) for the database containers to
 TIMEOUT=${TIMEOUT:-30}
 ```
 
-**Update your hosts file:**
+### Name resolution inside the docker network
 
-Add the following to your `/etc/hosts` file:
+This setup includes an instance of [DPS](http://mageddo.github.io/dns-proxy-server/latest/en/), a DNS Proxy server which takes care of container name resolution inside the docker network. All wikis at `*.web.mw.localhost` will resolve to the nginx-proxy container's IP address. The web container's name is `mediawiki.mw.localhost` and all other containers can be reached from each other via their corresponding host names:
```

Comment:
I'm not sure that this entire explanation is needed here. Think of a newbie developer or a volunteer who is just eager to start contributing to MW, this information might not be relevant to them, and might even cause more confusion than necessary. When writing tutorials or docs or presenting a technical subject I always find it useful to think about these points:

1. What is the "actionable" being described? What does the reader need to do to make / fix the thing they're trying to?
1. What is the default behaviour? Does describing it contribute to a layman's understanding, or is it just extra? If it's extra, but of value to more professional people, create a new page, and link to it through a "**_Learn more..._**"
1. Is there an alternate behaviour? If so, how simple it is to switch to it? Very simple: Add a "**_note_**" and describe the benefit, Not so simple? Add a page with precise and clear instructions and link to it through a "**_Learn more..._**"

Reply:
Thanks for your very helpful notes. I didn't add notes or any "Learn more..." pages, but simplified the explanations quite a lot. I also put the default option (_host file_) first now and the actionable alternative (_mount resolv.conf into DPS_) below.

Reply:
:100: Perfect, so much clearer now. Thank you :dancing_men: